### ISSUE TITLE
running on custom dataset

### ISSUE THREAD

**Issue by wangxianrui** [yihongXU/deepMOT#2]
if I want to run the deepmot on my own data, should i retrain the sameRPN network？

**Comment by yihongXU** [comment]
NO, you can have a try to run it on your own data set. But knowing that our implementation is based on data from MOTchallenge, so some modifications on the tracking code should be done in order to adapt to your data format.  And also, it is under "tracking by detections" paradigm (we need detections to initialize and terminate tracks), you need thus to provide detections for your dataset and the quality of your detections will have impact on the tracking performance.

**Comment by wangxianrui** [comment]
so, if i want  to track other objects with this  model , such as car, animial, and so on, what should i retrain ?
and another question,  i run the train script so slowly, may because you have not use the torch.data.Dataset,  will you consider update it ?

**Comment by yihongXU** [comment]
Hi, you should then finetune the SiamRPN with datasets having cars, animals, etc. 

Yes, you are right, because the training is frame by frame, and in a frame you could have more that 140 objects (i.e. MOT) and update the SiamRPN weights with MOT loss for this time step. There are several ways to improve the training speed:
remove the stupid for loop in SiamPRN_track, and input all the reference images and search images, (instead of one by one in a stupid loop),  as a batch into the tracking network, and this could be done by Dataset.
Recently, I will try to do it when I have free time.

**Comment by wangxianrui** [comment]
thanks a lot! if I run the train script on mot17 dataset, How long will it take?
thanks again,  thanks for your beautiful code!

**Comment by yihongXU** [comment]
well, the code is not beautiful enough...
It will around 72h to finetune on MOT17 on GTX Titan.

**Comment by wangxianrui** [comment]
but i have run 24h just 1 epoch on p40,  emmmm
have i do something wrong, i just run it as default...

**Comment by yihongXU** [comment]
Knowing that in MOT17, you have 3 different types of detections for one ground truth (even the dataset copies 3 times the same gts.). Some you are actually training one gt several times. 
You can add in line 53:
```python
if "SDP" not in vname:
    continue
```

**Comment by wangxianrui** [comment]
so, you say 72h just infer the training on the SDP？  i will try it
i just came into tracking, thanks for your patience

**Comment by wangaixue** [comment]
@wangxianrui  have you change the dataloader without for loop?
